pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -787,3 +788,65 @@ def longest_common_subsequence(seq1: OneDimensionalArray, seq2: OneDimensionalAr
                     check_mat[i][j] = check_mat[i][j-1]
 
     return OneDimensionalArray(seq1._dtype, check_mat[row][col][-1])
+
+def longest_increasing_subsequence(seq: OneDimensionalArray) -> int:
+    """
+    Finds the length of longest increasing subsequence of the given sequences.
+    Parameters
+    ========
+    seq: OneDimensionalArray
+        The sequence.
+    Returns
+    =======
+    output: int
+        The length of longest increasing subsequence.
+    Examples
+    ========
+    >>> from pydatastructs import longest_increasing_subsequence as LIS, OneDimensionalArray as ODA
+    >>> arr1 = [1, 3, 9, 4, 6]
+    >>> lis = LIS(arr1)
+    >>> str(lis)
+    '4'
+    >>> arr1 = [8, 7, 2, 12, 16, 28, 44]
+    >>> lis = LIS(arr1)
+    >>> str(lis)
+    '5'
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/Longest_increasing_subsequence
+
+    """
+    def CeilIndex(A, l, r, key):
+        while (r - l > 1):
+            m = l + (r - l)//2
+            if (A[m] >= key):
+                r = m
+            else:
+                l = m
+        return r
+
+    size = len(seq)
+    tailTable = [0 for i in range(size + 1)]
+    length = 0 # always points empty slot
+
+    tailTable[0] = seq[0]
```

Comment:
Use [snake case](https://en.wikipedia.org/wiki/Snake_case) and not [camel case](https://en.wikipedia.org/wiki/Camel_case).